web/packages/teleterm/src/mainProcess/agentRunner/agentRunner.ts

```diff
@@ -136,6 +157,17 @@ export class AgentRunner {
     process.once('error', errorHandler);
     process.once('exit', exitHandler);
   }
+
+  private updateProcessState(
+    rootClusterUri: RootClusterUri,
+    state: AgentProcessState
+  ): void {
+    const agent = this.agentProcesses.get(rootClusterUri);
+    if (agent) {
+      agent.state = state;
+    }
```

Comment:
When would we call `updateProcessState` with agent not being in the state?

Is it about `kill` and `killAll`? I feel like we might not need to delete the state of the agent from AgentRunner at all.

If we do want to remove them from the state, we should do so from functions which react to state changes (rather than those that initiate those changes by e.g. sending SIGKILL).

---

We have to keep this class as simple as possible because it'll get twice as complex after I add the agent cleanup daemon. 😏

Reply:
>When would we call updateProcessState with agent not being in the state?

I added this as a just-in-case check. 
I think we can leave the processes in the state, it doesn't change much (since killing already stopped process is noop) but simplifies things a bit.